clients/src/main/java/org/apache/kafka/common/requests/JoinGroupResponse.java

```diff
@@ -28,9 +28,15 @@ public class JoinGroupResponse extends AbstractResponse {
 
     private final JoinGroupResponseData data;
 
-    public JoinGroupResponse(JoinGroupResponseData data) {
+    public JoinGroupResponse(JoinGroupResponseData data, short version) {
         super(ApiKeys.JOIN_GROUP);
         this.data = data;
+
+        // All versions prior to version 7 do not support nullable
+        // string for the protocol type. Empty string should be used.
+        if (version < 7 && data.protocolType() == null) {
```

Comment:
I see that this was made nullable in version 7, but we are now on version 9. Did we just not enforce this before?

Reply:
Oh I see it used to be done here in KafkaApis:
```
        val protocolName = if (request.context.apiVersion() >= 7)
          joinResult.protocolName.orNull
        else
          joinResult.protocolName.getOrElse(GroupCoordinator.NoProtocol)
```

Reply:
Right it was there.